common/src/main/java/com/terraformersmc/biolith/impl/data/BiomePlacementMarshaller.java

```diff
@@ -117,27 +118,21 @@ public void unmarshall() {
         }
     }
 
-    public record AddSubBiomeMarshaller(RegistryKey<DimensionType> dimension, RegistryKey<Biome> target, RegistryKey<Biome> biome, SubBiomeMatcherMarshaller matcher) {
-        public static Codec<AddSubBiomeMarshaller> CODEC = RecordCodecBuilder.create(
-                (instance) -> instance.group(
-                                RegistryKey.createCodec(RegistryKeys.DIMENSION_TYPE).fieldOf("dimension")
-                                        .forGetter(AddSubBiomeMarshaller::dimension),
-                                RegistryKey.createCodec(RegistryKeys.BIOME).fieldOf("target")
-                                        .forGetter(AddSubBiomeMarshaller::target),
-                                RegistryKey.createCodec(RegistryKeys.BIOME).fieldOf("biome")
-                                        .forGetter(AddSubBiomeMarshaller::biome),
-                                SubBiomeMatcherMarshaller.CODEC.fieldOf("matcher")
-                                        .forGetter(AddSubBiomeMarshaller::matcher)
-                        )
-                        .apply(instance, AddSubBiomeMarshaller::new));
+    public record AddSubBiomeMarshaller(RegistryKey<DimensionType> dimension, RegistryKey<Biome> target, RegistryKey<Biome> biome, Criterion criterion) {
+        public static Codec<AddSubBiomeMarshaller> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+            RegistryKey.createCodec(RegistryKeys.DIMENSION_TYPE).fieldOf("dimension").forGetter(AddSubBiomeMarshaller::dimension),
+            RegistryKey.createCodec(RegistryKeys.BIOME).fieldOf("target").forGetter(AddSubBiomeMarshaller::target),
+            RegistryKey.createCodec(RegistryKeys.BIOME).fieldOf("biome").forGetter(AddSubBiomeMarshaller::biome),
+            Criterion.MATCHER_CODEC.fieldOf("criteria").forGetter(AddSubBiomeMarshaller::criterion)
```

Comment:
I don't much like it, but to be consistent with other practice in the PR, this field should be `criterion`.